server/sockets.js

```diff
@@ -1,12 +1,28 @@
 const socketio = require('socket.io');
+const getUserData = require('./database/get-user-data.js');
+const cookieParser = require('cookie');
 
-module.exports =  (listener) => {
+const createSocket = (listener) => {
   const io = socketio.listen(listener);
-  io.on('connection', (socket) => {
-    console.log("new connection");
+  let id = '';
+  io.set('authorization', (handshakeData, accept) => {
+
+    if (handshakeData.headers.cookie) {
+      id = cookieParser.parse(handshakeData.headers.cookie)['new-user'];
+    } else {
+      return accept('No cookie transmitted.', false);
+    }
+
+    accept(null, true);
+  });
 
-    socket.on('newGoals', (data) => {
-      console.log(data);
+  io.on('connection', (socket) => {
+    getUserData(id, (res) => {
```

Comment:
Also, I think it might be better to put the file 'format-user-dater' into `/server` as it technically doesn't interact with the database. It's kind of just a helper function. Maybe it can be in a `/helpers` folder instead or something...